### DISCUSSION TITLE
Editing and adding missing subtitles

### DISCUSSION THREAD

**Discussion by Rozekail** [hzqst/MetaHookSv#513] [General]
Hopefully this is the right place for this! I'm currently going through the entirety of the English subtitles and cleaning them up (fixing grammar/punctuation/spelling/etc.), as well as adding any missing subtitles I find, like the additional dialogue for the PS2 port's intro. I'm not sure what it takes to have that "approved" and added to the mod itself, but if that's something that anyone working on the mod would like, I'd be more than happy to provide the modified files once I'm done with them. 

My only concern at the moment is that certain subtitles don't display even though their corresponding audio files are properly set up in the mod's files (like H.E.V. suit dialogue), and my forté is language, not programming. But once I'm done, everything should be set for the English language. Unfortunately I can't contribute to any other languages but the ground work will be there so it should be much easier for someone to translate, at the very least.

Anyway, that's the end of my long-winded post. Cheers!

**Comment by hzqst** [comment]
You can utilize GitHub's fork feature by forking a copy of the MetaHookSv source code -> clone the forked repository to your local machine -> commit your subtitle changes to the \Build\svencoop\captionmod\dictionary_english.txt and Build\svencoop\captionmod\dictionary.csv files in your local source code directory -> submit your changes to GitHub -> initiate a Pull Request on GitHub, and then I can merge your changes into the main codebase. This way, your changes will be bundled into the release package the next time the MetaHookSv releases.

Here's a 1-minute video that can quickly guide you through the steps: https://www.youtube.com/watch?v=8lGpZkjnkt4

As for the issue of subtitles not displaying despite the audio files being configured, I suggest opening an issue specifically for that; in this way, I can promptly locate and fix the problem.

**Reply by Rozekail** [reply]
I'm very inexperienced with anything like GitHub so thank you for explaining it. I *should* have it done before the end of the year. And I'll open an issue, thanks again!

**Comment by Rozekail** [comment]
Upon making more progress, I've come across an odd (but should have been expected) quirk.

Certain types of formatting will cause subtitles to not display. For instance, in one of the PS2 Hazard Course lines, a scientist quotes a document that Gordon Freeman signed. That means that in the dialogue, that quote should have quotations around it. However, since quotations are what separate the different sections in the dictionary files, using them will invariably break the subtitles. 

The example:

"tr_sci4_legal" "Please listen carefully as there is a little paperwork to take care of before we get started. Oh, I see that you have already signed our legal disclaimer: "I, Gordon Freeman, hereby agree to the following terms, blah blah blah, in any case of serious injury, dismemberment, toxic poisoning, burns, rashes, lesions, blah blah, et cetera et cetera...hereby agree to waive all rights as an employee of the Black Mesa facility.""

This makes it so when you load into the Hazard Course, the first line of subtitles triggered will just display its name, not the dialogue. In the case of not having the PS2 Hazard Course mod installed, the first subtitle is for walking up to the holographic assistant, which will just display "#tr_holo_intro" instead of its relevant subtitle. 

This can be bypassed by using special quotation mark characters (like “ and ”, etc.). As long as they aren't the standard quotation marks you get when you press that key on a standard QWERTY keyboard ("), they're fine. 

Note: I have tried this with standard game subtitles, it isn't a mod causing some weird conflict (I don't even have the mod installed, I'm just putting in all the unused dialogue in case someone does have them installed). I didn't think this was an "issue" so much as just a limitation so I didn't want to open another issue for it, but thought it was odd enough that it should be mentioned. It took me way too long to figure out what I'd done wrong, but I honestly should have known.

**Reply by hzqst** [reply]
> Upon making more progress, I've come across an odd (but should have been expected) quirk.
> 
> Certain types of formatting will cause subtitles to not display. For instance, in one of the PS2 Hazard Course lines, a scientist quotes a document that Gordon Freeman signed. That means that in the dialogue, that quote should have quotations around it. However, since quotations are what separate the different sections in the dictionary files, using them will invariably break the subtitles.
> 
> The example:
> 
> "tr_sci4_legal" "Please listen carefully as there is a little paperwork to take care of before we get started. Oh, I see that you have already signed our legal disclaimer: "I, Gordon Freeman, hereby agree to the following terms, blah blah blah, in any case of serious injury, dismemberment, toxic poisoning, burns, rashes, lesions, blah blah, et cetera et cetera...hereby agree to waive all rights as an employee of the Black Mesa facility.""
> 
> This makes it so when you load into the Hazard Course, the first line of subtitles triggered will just display its name, not the dialogue. In the case of not having the PS2 Hazard Course mod installed, the first subtitle is for walking up to the holographic assistant, which will just display "#tr_holo_intro" instead of its relevant subtitle.
> 
> This can be bypassed by using special quotation mark characters (like “ and ”, etc.). As long as they aren't the standard quotation marks you get when you press that key on a standard QWERTY keyboard ("), they're fine.
> 
> Note: I have tried this with standard game subtitles, it isn't a mod causing some weird conflict (I don't even have the mod installed, I'm just putting in all the unused dialogue in case someone does have them installed). I didn't think this was an "issue" so much as just a limitation so I didn't want to open another issue for it, but thought it was odd enough that it should be mentioned. It took me way too long to figure out what I'd done wrong, but I honestly should have known.

special characters like quote can be correctly parsed and recognized by using `\"` instead of `"`·, in vgui2 localization fille.

**Reply by Rozekail** [reply]
Ah, excellent! I'm glad there's a way around it. Thanks!

**Comment by Rozekail** [comment]
I looked through the documentation but didn't see an answer so I'll just tack it on here. Is there a way to add certain types of formatting to subtitles, like italics/bold/underline/etc.? HTML code doesn't work, neither do your typical symbols like **** or __. It isn't really _necessary_, I'm just curious as it'd make it look a bit closer to the dialogue's intent.

**Comment by hzqst** [comment]
all subtitles sticks to one font:  https://github.com/hzqst/MetaHookSv/blob/e22843ab9ea40c2e539dde3b618f302dab219be3/Build/svencoop/captionmod/CaptionScheme.res#L83C3-L83C33
